src/Microsoft.ML.Transforms/NAIndicatorTransform.cs

```diff
@@ -68,322 +76,548 @@ private static VersionInfo GetVersionInfo()
+                }
+                else if (!sense && srcLength - indices.Count < srcLength / 2)
+                {
+                    // Will produce sparse output.
+                    int dstCount = srcLength - indices.Count;
+                    Utils.EnsureSize(ref dstValues, dstCount, false);
+                    Utils.EnsureSize(ref dstIndices, dstCount, false);
+
+                    // Appends the length of the src to make the loop simpler,
+                    // as the length of src will never be reached in the loop.
+                    indices.Add(srcLength);
+
+                    int iiDst = 0;
+                    int iiSrc = 0;
+                    int iNext = indices[iiSrc];
+                    for (int i = 0; i < srcLength; i++)
+                    {
+                        Host.Assert(0 <= i && i <= iNext);
+                        Host.Assert(iiSrc + iiDst == i);
+                        if (i < iNext)
+                        {
+                            Host.Assert(iiDst < dstCount);
+                            dstValues[iiDst] = true;
+                            dstIndices[iiDst++] = i;
+                        }
+                        else
+                        {
+                            Host.Assert(iiSrc + 1 < indices.Count);
+                            Host.Assert(iNext < indices[iiSrc + 1]);
+                            iNext = indices[++iiSrc];
+                        }
+                    }
+                    Host.Assert(srcLength == iiSrc + iiDst);
+                    Host.Assert(iiDst == dstCount);
+
+                    dst = new VBuffer<bool>(srcLength, dstCount, dstValues, dstIndices);
+                }
+                else
+                {
+                    // Will produce dense output.
+                    Utils.EnsureSize(ref dstValues, srcLength, false);
+
+                    // Appends the length of the src to make the loop simpler,
+                    // as the length of src will never be reached in the loop.
+                    indices.Add(srcLength);
+
+                    int ii = 0;
+                    for (int i = 0; i < srcLength; i++)
+                    {
+                        Host.Assert(0 <= i && i <= indices[ii]);
+                        if (i == indices[ii])
+                        {
+                            dstValues[i] = sense;
+                            ii++;
+                            Host.Assert(ii < indices.Count);
+                            Host.Assert(indices[ii - 1] < indices[ii]);
+                        }
+                        else
+                            dstValues[i] = !sense;
+                    }
+
+                    dst = new VBuffer<bool>(srcLength, dstValues, dstIndices);
+                }
+            }
         }
+    }
+
+    public sealed class NAIndicatorEstimator : TrivialEstimator<NAIndicatorTransform>
+    {
+        private readonly (string input, string output)[] _columnPairs;
```

Comment:
>_columnPairs [](start = 57, length = 12)

OneToOneTransformerBase has a property called ColumnPairs. This can be used in GetOutputSchema instead of storing this on the Estimator class.

Reply:
I think that it is not accessible from the NAIndicatorEstimator.

---
In reply to: [225365611](https://github.com/dotnet/machinelearning/pull/1217#discussion_r225365611) [](ancestors = 225365611)

Reply:
This is because you need to make a public property on the transformer that exposes `ColumnPairs`.

---
In reply to: [225367166](https://github.com/dotnet/machinelearning/pull/1217#discussion_r225367166) [](ancestors = 225367166,225365611)